### ISSUE TITLE
Possible to make keyboard shortcut for this action?

### ISSUE THREAD

**Issue by rrthomas** [p91paul/middleclickclose#13]
I don't know if this would require a big change, but it'd be really cool to have this function on the keyboard… I have got used to navigating the Overview with the keyboard, and I was just looking for a way to close the currently highlighted window directly (rather than select it then close).

**Comment by zhonghuabao1** [comment]
+1,
@rrthomas have you get the way? it would be cool!

**Comment by rrthomas** [comment]
No, sadly.

**Comment by gtsiam** [comment]
It is now possible (in the master branch at least). Invoking the `close` action (typically `Alt+F4`) in the overview will now close the focused window.

However, do note: there's some bizarre logic where the workspace you're looking at is not actually "active" (whatever that means). Though unraveling that mess is best left for a proper gnome-shell patch. For now, I'm emulating the behavior of the `Enter` key, which will switch to the "active" workspace whenever gnome-shell gets confused.

Please do [download](https://github.com/user-attachments/files/16143035/middleclickclose%40paolo.tranquilli.gmail.com.shell-extension.zip) it and test it out. I am slightly scared to send this to EGO without proper testing (potential bugs include accidentally closing windows in another workspace), so I'll be running it myself for a week or so before I submit it for review

Also I'm waiting for the GNOME 47 extension migration guide to come out. It's running just fine on 47.alpha, but you never know.

TODO:
- [ ] Better documentation
- [ ] Add toggle for this behaviour

**Comment by rrthomas** [comment]
@gtsiam Thanks for working on this issue! Sorry, I'm not using this extension at present.

**Comment by gtsiam** [comment]
> @gtsiam Thanks for working on this issue! Sorry, I'm not using this extension at present.

Well it's only been like 7 years 😄

**Comment by rrthomas** [comment]
I have fixed issues after *much* longer than that myself!

**Comment by gtsiam** [comment]
> Does anyone here know how to navigate the overview across multiple monitors? The arrow keys only work within one monitor for me, I have no clue how to get to the other without using a mouse. (Gnome 45)

You're probably looking for the `Ctrl`+`Alt`+`Left/Right` arrow keys.

**Comment by dreua** [comment]
> You're probably looking for the `Ctrl`+`Alt`+`Left/Right` arrow keys.

Nope, that just switches to another workspace but I have multiple physical monitors (or screens?) in one one workspace. Maybe this use case is just not really supported by gnome? I mean I don't mind, its just that I wanted to try this feature but without the ability to navigate to all windows with the keyboard only it doesn't make sense to me :shrug:

**Comment by gtsiam** [comment]
> Nope, that just switches to another workspace but I have multiple physical monitors

Oh, I can't read 🤦🏽.

> Maybe this use case is just not really supported by gnome?

Doesn't seem to be. But it _could_ be - I can take a look at gnome's source, but no promises.

**Comment by gtsiam** [comment]
Closing this, as it's implemented in v31.
